crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -107,6 +110,138 @@ impl From<PersistedQueriesPublishInput> for QueryVariables {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RelayPersistedQueryManifest {
+    #[serde(flatten)]
+    operations: HashMap<String, String>,
+}
+
+impl TryFrom<RelayPersistedQueryManifest> for ApolloPersistedQueryManifest {
+    type Error = RoverClientError;
+
+    fn try_from(relay_manifest: RelayPersistedQueryManifest) -> Result<Self, Self::Error> {
+        if relay_manifest.operations.is_empty() {
+            return Err(RoverClientError::NoRelayOperations);
+        }
+
+        let mut anonymous_operations = Vec::new();
+        let mut syntax_errors = Vec::new();
+        let mut ids_with_multiple_operations = Vec::new();
+        let mut ids_with_no_operations = Vec::new();
+        let mut operations = Vec::new();
+        for (id, body) in relay_manifest.operations {
+            let ast = Parser::new(&body).parse();
+
+            let definitions: Vec<Definition> = ast.clone().document().definitions().collect();
+
+            let maybe_definition = match definitions.len() {
+                0 => {
+                    ids_with_no_operations.push(id.clone());
+                    None
+                }
+                1 => {
+                    if let Some(Definition::OperationDefinition(definition)) = definitions.get(0) {
+                        Some(definition)
+                    } else {
+                        ids_with_no_operations.push(id.clone());
+                        None
+                    }
+                }
+                _ => {
+                    ids_with_multiple_operations.push(id.clone());
```

Comment:
and this is complaining if there is more than one. will this not choke on "one operation definition, one fragment"? seems like we need to filter `definitions` down to just operation definitions up here rather than below (or do the error checking more "lazily" as you iterate rather than by an up-front count)